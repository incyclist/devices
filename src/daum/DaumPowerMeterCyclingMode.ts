import CyclingMode, { UpdateRequest} from "../CyclingMode";
import PowerMeterCyclingMode from "../modes/power-meter";


export default class DaumPowerMeterCyclingMode extends PowerMeterCyclingMode implements CyclingMode {

    prevRequest: UpdateRequest;
    hasBikeUpdate: boolean = false;
    
    getBikeInitRequest(): UpdateRequest {
        return {slope:0}    
    }    

    sendBikeUpdate(request: UpdateRequest): UpdateRequest {
        super.sendBikeUpdate(request);
        this.prevRequest = {}
        return {}
    }

}