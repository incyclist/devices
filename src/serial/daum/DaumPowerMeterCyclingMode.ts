import CyclingMode, { UpdateRequest} from "../../modes/cycling-mode";
import PowerMeterCyclingMode from "../../modes/power-meter";


export default class DaumPowerMeterCyclingMode extends PowerMeterCyclingMode implements CyclingMode {

    prevRequest: UpdateRequest;
    hasBikeUpdate: boolean = false;
    
    getBikeInitRequest(): UpdateRequest {
        return {slope:0}    
    }    

    sendBikeUpdate(request: UpdateRequest): UpdateRequest {
        super.sendBikeUpdate(request);
        this.prevRequest = {}
        return {}
    }

}