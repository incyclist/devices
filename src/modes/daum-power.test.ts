import DaumPowerMeterCyclingMode from "./daum-power"
import MockAdapter from '../../test/mock-adapter';

describe('DaumPower',()=>{
    test('statics',()=>{
        const m = new DaumPowerMeterCyclingMode(new MockAdapter())

        expect(m.getName()).toBe('PowerMeter')
        expect(m.getDescription()).toMatchSnapshot()
        expect(m.getProperties()).toEqual([])
        expect(DaumPowerMeterCyclingMode.supportsERGMode()).toBe(false)        
                
    })

    test('init request is always slope:0',()=>{
        const m = new DaumPowerMeterCyclingMode(new MockAdapter())
        const res = m.getBikeInitRequest()
        expect(res).toMatchObject({slope:0})
    })

})