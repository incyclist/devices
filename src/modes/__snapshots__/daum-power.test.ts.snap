// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaumPower statics 1`] = `
"Power and cadence are taken from device. Speed is calculated from power and current slope
This mode will not respect maximum power and/or workout limits"
`;
